### PR TITLE
Update printfn/sprintf to interpolations

### PR DESCRIPTION
**Pull request by cartermp**
For https://github.com/dotnet/docs/issues/21272

### CODE REVIEW

File: docs/fsharp/introduction-to-functional-programming/index.md [hunk 1]
```diff
@@ -152,7 +152,7 @@ Here is another example of a non-pure function, because it performs a side effec
 
 ```fsharp
 let addOneToValue x =
-    printfn "x is %d" x
+    printfn $"x is %d{x}" x
```
Comment (slang25) [inline docs/fsharp/introduction-to-functional-programming/index.md:155]:
```suggestion
    printfn $"x is %d{x}"
```

File: docs/fsharp/tutorials/type-providers/creating-a-type-provider.md [hunk 2]
```diff
@@ -347,7 +347,7 @@ t.AddMembersDelayed(fun () ->
                   getterCode= (fun args -> <@@ valueOfTheProperty @@>))
 
               p.AddXmlDocDelayed(fun () ->
-                  sprintf "This is StaticProperty%d on NestedType" i)
+                  sprintf $"This is StaticProperty{i} on NestedType")
```
Comment (slang25) [inline docs/fsharp/tutorials/type-providers/creating-a-type-provider.md:350]:
```suggestion
                  $"This is StaticProperty{i} on NestedType")
```

**Comment by cartermp** [comment]
Thanks for the suggestions @slang25 !

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Pull request closed without merging**